packages/lib/ren-tx/src/configs/genericBurn.ts

```diff
@@ -113,128 +106,175 @@ const burnTransactionListener = (context: BurnMachineContext) => (
+                    burnRef._cancel();
+                    burnRef.removeListener("confirmation", burnListener);
                 });
 
                 try {
-                    const res = await releaseRef
-                        .on("status", releaseListener)
-                        .on("transaction", transactionListener)
-                        .on("txHash", hashListener);
-                    callback({
-                        type: "RELEASED",
-                        data: { ...tx, renResponse: res },
-                    });
-                } catch (e) {
-                    callback({ type: "RELEASE_ERROR", data: new Error(e) });
+                    const r = await burnRef
+                        // host chain tx hash
+                        .on("transactionHash", (txHash: string) => {
+                            tx = {
+                                sourceTxHash: txHash,
+                                sourceTxConfs: 0,
+                                sourceTxAmount: Number(
+                                    context.tx.suggestedAmount,
+                                ),
+                                rawSourceTx: {
+                                    amount: String(context.tx.suggestedAmount),
+                                    transaction: {},
+                                },
+                            };
+                            callback({
+                                type: "SUBMITTED",
+                                data: {
+                                    ...tx,
+                                },
+                            });
+                        })
+                        .on("confirmation", burnListener);
+                    if (tx) {
+                        // the burn status is canonical, we should proceed if
+                        // it says we have burned
+                        if (r.status == BurnAndReleaseStatus.Burned) {
+                            // FIXME: ensure that the confs always match
+                            // the target
+                            tx.sourceTxConfs = 7;
+                        }
+
+                        // Always call because we won't get an emission
+                        // if the burn is already done
+                        burnListener(tx.sourceTxConfs);
+                    }
+                } catch (error) {
+                    throw error;
                 }
             };
 
-            receive((event: BurnMachineEvent) => {
+            receive(async (event: BurnMachineEvent) => {
                 if (event.type === "SUBMIT") {
                     performBurn()
                         .then()
                         .catch((e) => {
                             console.error(e);
-                            callback({ type: "BURN_ERROR", data: e });
+                            callback({
+                                type: "BURN_ERROR",
+                                data: e.toString(),
```

Comment:
What was the reason for changing the errors from Errors to strings? Wouldn't having the full error stack be potentially helpful for developers?

Reply:
I'll double check - but it seems like the errors were getting stripped in xstate unless they are stringified. The error needs to be serializable anyhow, as the intention is to have it land in persistence so that it can be reviewed